runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -197,7 +197,11 @@ private void rebuildAsync(boolean updated, boolean skillPaused, XpSnapshotSingle
 			// Update information labels
 			expGained.setText(htmlLabel("XP Gained: ", xpSnapshotSingle.getXpGainedInSession()));
 			expLeft.setText(htmlLabel("XP Left: ", xpSnapshotSingle.getXpRemainingToGoal()));
-			actionsLeft.setText(htmlLabel(xpSnapshotSingle.getActionType().getLabel() + ": ", xpSnapshotSingle.getActionsRemainingToGoal()));
+			// Note that slayer is a special case since it works like an EXPERIENCE skill but should be labeled as
```

Comment:
I would probably not special case slayer because 1. it is ugly, 2. you need to special case it everywhere where the label is used

Reply:
should be addressed by d18bb493b2443d381984f40440cb466c94d78936 - refactored the action type into two parts the label and the task (so now label can be either actions or kills and the task can be either experience or actor health). now it still tracks actions for experience vs actor health separately but the label for those actions can be changed to actions or kills whenever needed. So this makes it possible to set the label for slayer's experience task type to kills in the onExperienceChanged method.